lib/taskana-core/src/test/java/acceptance/workbasket/CreateWorkbasketAccTest.java

```diff
@@ -110,59 +109,45 @@ public void testCreateWorkbasketWithMissingRequiredField()
         workbasket.setName("Megabasket");
         workbasket.setType(WorkbasketType.GROUP);
         workbasket.setOrgLevel1("company");
-        try { // missing key
-            workbasketService.createWorkbasket(workbasket);
-            fail("InvalidWorkbasketException was expected");
-        } catch (InvalidWorkbasketException e) {
-        }
-
-        workbasket = workbasketService.newWorkbasket("key", "novatec");
-        workbasket.setType(WorkbasketType.GROUP);
-        workbasket.setOrgLevel1("company");
-        try {  // missing name
-            workbasketService.createWorkbasket(workbasket);
-            fail("InvalidWorkbasketException was expected");
-        } catch (InvalidWorkbasketException e) {
-        }
-
-        workbasket = workbasketService.newWorkbasket("key", "novatec");
-        workbasket.setName("Megabasket");
-        workbasket.setOrgLevel1("company");
-        try {  // missing type
-            workbasketService.createWorkbasket(workbasket);
-            fail("InvalidWorkbasketException was expected");
-        } catch (InvalidWorkbasketException e) {
-        }
-
-        workbasket = workbasketService.newWorkbasket("key", null);
-        workbasket.setName("Megabasket");
-        workbasket.setType(WorkbasketType.GROUP);
-        workbasket.setOrgLevel1("company");
-        try {  // missing domain
-            workbasketService.createWorkbasket(workbasket);
-            fail("InvalidWorkbasketException was expected");
-        } catch (InvalidWorkbasketException e) {
-        }
-
-        workbasket = workbasketService.newWorkbasket("", "novatec");
-        workbasket.setName("Megabasket");
-        workbasket.setType(WorkbasketType.GROUP);
-        workbasket.setOrgLevel1("company");
-        try {  // empty key
-            workbasketService.createWorkbasket(workbasket);
-            fail("InvalidWorkbasketException was expected");
-        } catch (InvalidWorkbasketException e) {
-        }
-
-        workbasket = workbasketService.newWorkbasket("key", "novatec");
-        workbasket.setName("");
-        workbasket.setType(WorkbasketType.GROUP);
-        workbasket.setOrgLevel1("company");
-        try {  // empty name
-            workbasketService.createWorkbasket(workbasket);
-            fail("InvalidWorkbasketException was expected");
-        } catch (InvalidWorkbasketException e) {
-        }
+        // missing key
+        Assertions.assertThrows(InvalidWorkbasketException.class, () -> workbasketService.createWorkbasket(workbasket));
+
+        Workbasket workbasket2 = workbasketService.newWorkbasket("key", "novatec");
```

Comment:
why are you not reuising the variable workbasket? I think the number in the name is not aesthetically pleasing.

Reply:
The test itself is not pleasing and covers way too much ... but the technical reason is that you can only use final or effectively final variables within lambda expressions.